crates/extensions/c8y_mapper_ext/src/operations/config_snapshot.rs

```diff
@@ -189,60 +181,62 @@ impl CumulocityConverter {
             Some(command.into_generic_command(&self.mqtt_schema)),
         ))
     }
+}
 
-    /// Resumes `config_snapshot` operation after required file was downloaded
-    /// from the File Transfer Service.
-    pub async fn handle_fts_config_download_result(
-        &mut self,
-        cmd_id: CmdId,
-        download_result: DownloadResult,
-        fts_download: FtsDownloadOperationData,
-    ) -> Result<Vec<MqttMessage>, ConversionError> {
-        let topic_id = fts_download.entity_topic_id;
-        let smartrest_topic = self.smartrest_publish_topic_for_entity(&topic_id)?;
-        let payload = fts_download.message.payload_str()?;
-        let response = &ConfigSnapshotCmdPayload::from_json(payload)?;
-
-        let download = match download_result {
-            Err(err) => {
-                let smartrest_error =
-                    fail_operation(
-                        CumulocitySupportedOperations::C8yUploadConfigFile,
-                        &format!("tedge-mapper-c8y failed to download configuration snapshot from file-transfer service: {err}"),
-                    );
+pub fn topic_filter(mqtt_schema: &MqttSchema) -> TopicFilter {
```

Comment:
Minor: It would have been better to leave this function and the next `impl CumulocityConverter` blocks in their former location (at the top) would have reduced the diff volume and better highlighted the changed bits. The **request** mapping logic appearing first looks more intuitive as well.

Reply:
Indeed, the diff looks much easier to review. I'll remember not to shuffle the code next time for easier review.
fixed in b8bd269